onnxruntime/contrib_ops/cpu/word_conv_embedding.cc

```diff
@@ -0,0 +1,219 @@
+  int64_t conv_res_segment_size = unfolded_width * num_filters;
+  int64_t memcpy_size = unfolded_kernal_size * sizeof(float);
+
+  auto input_unfolded_buffer_p = IAllocator::MakeUniquePtr<float>(allocator, seq_len * unfolded_segment_size);
+  auto conv_result_p = IAllocator::MakeUniquePtr<float>(allocator, seq_len * conv_res_segment_size);
+  auto conv_activation_result_p = IAllocator::MakeUniquePtr<float>(allocator, seq_len * conv_res_segment_size);
+
+  for (int64_t word_inx = 0; word_inx < seq_len; word_inx++) {
+    if (words_len_ptr[word_inx] <= 0) continue;
+
+    const float* current_word_input = input + word_inx * input_word_size;
+    float* current_word_unfolded_buffer_p = input_unfolded_buffer_p.get() + word_inx * unfolded_segment_size;
+    float* conv_buf_p = conv_result_p.get() + word_inx * conv_res_segment_size;
+    float* pactivationbuf = conv_activation_result_p.get() + word_inx * conv_res_segment_size;
+    float* pres = output + word_inx * num_filters;
+
+    // Unfolding from pin to pufbuf.
+    float* tmp_unfolded_buffer_ptr = current_word_unfolded_buffer_p;
+    for (int64_t unfolded_inx = 0; unfolded_inx < unfolded_width; unfolded_inx++) {
+      memcpy(tmp_unfolded_buffer_ptr, current_word_input, memcpy_size);
+      current_word_input += char_embedding_size;
+      tmp_unfolded_buffer_ptr += unfolded_kernal_size;
+    }
+
+    // Matrix multiplication.
+    //mkl_set_num_threads_local( 1 );
+    math::GemmEx<float, CPUMathUtil>(
+        CblasNoTrans, CblasTrans,
+        static_cast<int>(unfolded_width), static_cast<int>(num_filters), static_cast<int>(unfolded_kernal_size), 1.0f,
+        current_word_unfolded_buffer_p, static_cast<int>(unfolded_kernal_size),
+        weights, static_cast<int>(unfolded_kernal_size), 0.0f,
+        conv_buf_p, static_cast<int>(num_filters), &CPUMathUtil::Instance());
+    //MatrixMult( current_word_unfolded_buffer_p, weights, conv_buf_p, unfolded_width, num_filters, unfolded_kernal_size );
+
+    for (int64_t unfolded_inx = 0; unfolded_inx < unfolded_width; unfolded_inx++)
+      for (int64_t filter_inx = 0; filter_inx < num_filters; filter_inx++) {
+        conv_buf_p[unfolded_inx * num_filters + filter_inx] += bias[filter_inx];
+      }
+
+    MlasComputeTanh(conv_buf_p, pactivationbuf, unfolded_width * num_filters);
+
+    // Max pooling.
+    for (int64_t filter_inx = 0; filter_inx < num_filters; filter_inx++) {
+      pres[filter_inx] = -1.0f * 1e12f;
+    }
+
+    for (int64_t unfolded_inx = 0; unfolded_inx < unfolded_width; unfolded_inx++) {
+      if (unfolded_inx > 0 && unfolded_inx > (words_len_ptr[word_inx] - filter_width)) break;
+      float* pcur = (float*)pactivationbuf + unfolded_inx * num_filters;
+      for (int64_t filter_inx = 0; filter_inx < num_filters; filter_inx++) {
+        pres[filter_inx] = std::max(pcur[filter_inx], pres[filter_inx]);
+      }
+    }
+  }
+}
+void WordConvEmbedding::CalculateSuquenceWordsLength(
```

Comment:
suquence->sequence

Reply:
This functions calculate the length of each words in the sequence.

Reply:
My comment was that the function name appears to be a typo. What is suquence?

Reply:
Maybe the name CalculateLengthOfSequenceOfWords is better.